.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,92 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build-and-test:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out repository code (from PR).
+      uses: actions/checkout@v2.4.0
+
+    - name: Install nix.
+      uses: cachix/install-nix-action@v15
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+      # Would using cachix help in anyway? I think we are not building anything, we're
+      # only fetching things from cache.nixos.org and iohk.cachix.org
+      # - uses: cachix/cachix-action@v10
+      #   with:
+      #     name: plutus-libs
+      #     authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+      
+    - name: Accessing the cabal cache.
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cabal/store
+          dist-newstyle
+        key: cabal-cache-${{ github.sha }}
+        restore-keys: |
+          cabal-cache
+
+    - name: Accessing the result cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./*.out
+          ./*.res
+        key: result-cache-${{ github.sha }}
+
+    - name: Chmod test script.
+      run: |
+        chmod u+x ci/run-tests.sh
+
+    - name: Build and test pirouette
+      uses: ./.github/actions/nix-run
+      with:
+        command: ./ci/run-tests.sh --ci
+
+  check-result:
+    needs: build-and-test
+    name: check-result
+    runs-on: ubuntu-latest
+    steps:
+    - name: Accessing the result cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./*.out
+          ./*.res
+        key: result-cache-${{ github.sha }}
+
+    - name: Check result
+      run: |
+        is_ok=true
```

Comment:
Maybe extract this to a script? I am not sure whether this is even possible in GH Actions, though.

Reply:
it is possible, but it would mean we need to checkout the repo code again in the other job